src/main/kotlin/com/doist/gradle/changelog/ChangelogProcessor.kt

```diff
@@ -0,0 +1,74 @@
+package com.doist.gradle.changelog
+
+import java.io.File
+import java.io.RandomAccessFile
+import java.util.LinkedList
+
+internal class ChangelogProcessor(private val config: CommitConfig) {
+    fun collectPendingEntries(file: File): List<String> {
+        return file.walk()
+            .filter { it.isFile }
+            .map { it.readLines() }
+            .flatten()
+            .filter { it.isNotBlank() }
+            .toList()
+    }
+
+    fun insertEntries(changelogFile: File, entries: List<String>) {
+        if (entries.isEmpty()) return
+
+        val changelog = LinkedList(changelogFile.readLines())
```

Comment:
I think it's not optimal to read file several times. Wouldn't it be better to read the file once with `File.readText()` as string, then find line separator in the text and split this string by this line separator. Given that `readText` doesn't skip the last empty line, we won't need additional checks on it :)

Reply:
Yes, that's a much better solution 🤗 Updated.